### PR TITLE
feat(release process): resume to last edited page

### PR DESCRIPTION
**Pull request by kunalgaurav-bmw**
## Description

Resume to the last edited page during App/Service release process.

Changelog entry:

```
- **App & Service Release Process**:
  - enabled resuming to the last edited page during App/Service release process. [#1376](https://github.com/eclipse-tractusx/portal-frontend/pull/1376)
```

## Why

Currently if I stop the release process and then get back to the app/service I need to click through all the steps I fulfilled already, It should be automatically resume to the last page where user stop. 

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/1335

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

**Review by evegufy — COMMENTED** [review]
@kunalgaurav-bmw please see https://github.com/eclipse-tractusx/portal-frontend/issues/1335#issuecomment-2516533950
@manojava-gk @lavanya-bmw please review

**Comment by manojava-gk** [comment]
@kunalgaurav-bmw its not registration process -> it is release process. Pls change the text in both title and description

**Comment by manojava-gk** [comment]
@kunalgaurav-bmw could you pls confirm whether only mandatory fields are being checked to increment or decrement the step?

**Comment by kunalgaurav-bmw** [comment]
> @kunalgaurav-bmw could you pls confirm whether only mandatory fields are being checked to increment or decrement the step?

Hi @manojava-gk  I have done this changes in useEffect in the following way for the all steps and all the mandatory fields I have added in if condition

useEffect(() => {
    if (hasDispatched.current) return
    if (
      fetchAppStatus?.title &&
      fetchAppStatus.provider &&
      fetchAppStatus.descriptions[0]?.shortDescription &&
      fetchAppStatus.descriptions[1]?.shortDescription &&
      fetchAppStatus.useCase?.length &&
      fetchAppStatus.supportedLanguageCodes?.length &&
      fetchAppStatus.documents?.APP_LEADIMAGE?.length &&
      appRedirectStatus
    ) {
      dispatch(increment())
      hasDispatched.current = true
    }
  }, [fetchAppStatus, hasDispatched])

**Comment by manojava-gk** [comment]
@kunalgaurav-bmw Is it possible to determine the step based on response in the first step itself?

Create a helper file and move all the step specific conditions to the respective places,
```
function determineStep() {
if(firstStep)
 -> increment(1)
if(secondStep)
 -> increment(2)
 }
 
 function firstStep() {}
 function secondStep() {}
 
 like this ...
 ```

component will have only one condition in first step to determine the actual step. All the step based condition will be hosted in the main method as shown above

File: src/components/pages/AppOverview/index.tsx [hunk 1]
```diff
@@ -240,6 +243,12 @@ export default function AppOverview() {
     }
   }
 
+  useEffect(() => {
+    return () => {
+      dispatch(setAppRedirectStatus(true))
+    }
+  }, [])
```
Comment (oyo) [inline src/components/pages/AppOverview/index.tsx:250]: I haven't seen this pattern in our code before. This returns a cleanup function as described in https://react.dev/reference/react/useEffect#parameters - if the dependency array is empty what difference does it make to not simply call the function but return an internal function?

```
  useEffect(() => {
      dispatch(setAppRedirectStatus(true))
  }, [])
```

File: src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx [hunk 2]
```diff
@@ -197,6 +200,12 @@ export default function ServiceListOverview() {
     dispatch(setServiceStatus(initialState.serviceStatusData))
   }
 
+  useEffect(() => {
+    return () => {
+      dispatch(setServiceRedirectStatus(true))
+    }
+  }, [])
+
```
Comment (oyo) [inline src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx:208]: another cleanup function - see my other comment. can you explain the need for that?

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx [hunk 3]
```diff
@@ -216,6 +219,24 @@ export default function AppMarketCard() {
...
+      fetchAppStatus.useCase?.length &&
+      fetchAppStatus.supportedLanguageCodes?.length &&
+      fetchAppStatus.documents?.APP_LEADIMAGE?.length &&
+      fetchAppStatus.price &&
+      appRedirectStatus
+    ) {
+      dispatch(increment())
+      hasDispatched.current = true
+    }
+  }, [fetchAppStatus, hasDispatched])
```
Comment (oyo) [inline src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx:238]: This seems quite complicated and hard to follow. What issue are you trying to solve here?

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 4]
```diff
@@ -165,6 +169,20 @@ export default function AppPage() {
...
+      fetchAppStatus.descriptions[1]?.longDescription &&
+      fetchAppStatus.images?.length &&
+      fetchAppStatus?.privacyPolicies?.length &&
+      appRedirectStatus
+    ) {
+      dispatch(increment())
+      hasDispatched.current = true
+    }
+  }, [fetchAppStatus, hasDispatched])
+
```
Comment (oyo) [inline src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:185]: another complicated block

**Review by oyo — CHANGES_REQUESTED** [review]
I'm unable to follow what's happening here - guess we should discuss this in a call.

File: src/components/pages/AppOverview/index.tsx (hunk 1 above)
Comment (kunalgaurav-bmw) [reply src/components/pages/AppOverview/index.tsx:250]: Explained

File: src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx (hunk 2 above)
Comment (kunalgaurav-bmw) [reply src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx:208]: Explained

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx (hunk 4 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:185]: Explained all the mandatory fields I have added in if condition.

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx (hunk 3 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx:238]: Explained all the mandatory fields I have added in if condition.

**Review by manojava-gk — APPROVED** [review]

**Review by lavanya-bmw — APPROVED** [review]
lgtm

File: src/components/pages/AppOverview/index.tsx (hunk 1 above)
Comment (evegufy) [reply src/components/pages/AppOverview/index.tsx:250]: show how, refer to commit or code section

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx (hunk 3 above)
Comment (evegufy) [reply src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx:232]: show how, refer to commit or code section

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx (hunk 4 above)
Comment (evegufy) [reply src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:183]: show how, refer to commit or code section

File: src/components/pages/AppOverview/index.tsx (hunk 1 above)
Comment (kunalgaurav-bmw) [reply src/components/pages/AppOverview/index.tsx:250]: I have created two helper class AppStepHelper and offerStepHelper to manage the steps and here resetting the status on landing AppOverview page please refer to commit [8473552](https://github.com/eclipse-tractusx/portal-frontend/pull/1376/commits/847355214828237573e02757cc7adf45c0e25313)

File: src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx (hunk 2 above)
Comment (kunalgaurav-bmw) [reply src/components/pages/ServiceReleaseProcess/components/ServiceListOverview.tsx]: I have created helper class  offerStepHelper to manage the steps and here reseting the status on landing ServiceListOverview page please refer to commit [8473552](https://github.com/eclipse-tractusx/portal-frontend/pull/1376/commits/847355214828237573e02757cc7adf45c0e25313)

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx (hunk 3 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx:232]: I handled this in helper class AppStepHelper to manage the steps using required fields please refer to commit  [8473552](https://github.com/eclipse-tractusx/portal-frontend/pull/1376/commits/847355214828237573e02757cc7adf45c0e25313)

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx (hunk 4 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:183]: I handled this in helper class AppStepHelper to manage the steps using required fields please refer to commit  [8473552](https://github.com/eclipse-tractusx/portal-frontend/pull/1376/commits/847355214828237573e02757cc7adf45c0e25313)

File: src/components/shared/basic/ReleaseProcess/AppStepHelper/index.tsx
Comment (evegufy) [inline src/components/shared/basic/ReleaseProcess/AppStepHelper/index.tsx:1]: add file header

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/components/shared/basic/ReleaseProcess/OfferStepHelper/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { ConsentStatusEnum } from 'features/serviceManagement/apiSlice'
```
Comment (evegufy) [inline src/components/shared/basic/ReleaseProcess/OfferStepHelper/index.tsx:1]: add file header

**Review by evegufy — CHANGES_REQUESTED** [review]

File: src/components/shared/basic/ReleaseProcess/AppStepHelper/index.tsx
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppStepHelper/index.tsx:1]: file header added

File: src/components/shared/basic/ReleaseProcess/OfferStepHelper/index.tsx (hunk 5 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/OfferStepHelper/index.tsx:20]: file header added

**Review by evegufy — APPROVED** [review]
